test/suite/test_dump04.py

```diff
@@ -0,0 +1,96 @@
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wttest
+from suite_subprocess import suite_subprocess
+
+# test_dump04.py
+# test dump utility with -j and -k options
+class test_dump04(wttest.WiredTigerTestCase, suite_subprocess):
+    output = "dump.out"
+
+    def wrap_in_json(self, s1, s2):
+        return f"\"{s1}\" : \"{s2}\""
+
+    def key_to_json(self, s):
+        return self.wrap_in_json("key0", s)
+
+    def value_to_json(self, s):
+        return self.wrap_in_json("value0", s)
+
+    def string_to_unicode(self, s):
+        ret = ""
+        for c in s:
+            ret += '\\u{:04x}'.format(ord(c))
+        return ret
+
+    def test_dump(self):
+        uri = 'table:test_dump'
+        create_params = 'key_format=u,value_format=u'
+        self.session.create(uri, create_params)
+
+        cursor = self.session.open_cursor(uri)
+        for i in range(1, 5):
+            cursor[str(i)] = str(i)
+        cursor['key'] = 'value'
+        cursor.close()
+
+        # Perform checkpoint, to clean the dirty pages and place values on disk.
+        self.session.checkpoint()
+
+        # Call dump with -j option
+        self.runWt(['dump', '-j', uri], outfilename=self.output)
+        self.check_file_contains(self.output, self.key_to_json(self.string_to_unicode("key")))
+        self.check_file_contains(self.output, self.value_to_json(self.string_to_unicode("value")))
+        for i in range(1, 5):
+            self.check_file_contains(self.output, self.key_to_json(self.string_to_unicode(str(i))))
+            self.check_file_contains(self.output, self.value_to_json(self.string_to_unicode(str(i))))
+
+        # Call dump with -k option
+        self.runWt(['dump', '-k', "key", uri], outfilename=self.output)
+        self.check_file_contains(self.output, "key\n")
+        self.check_file_contains(self.output, "value\n")
+        for i in range(1, 5):
+            self.check_file_not_contains(self.output, f"{i}\n")
+            self.check_file_not_contains(self.output, f"{i}\n")
+
+        # Call dump with -j and -k options
+        self.runWt(['dump', '-j', '-k', "key", uri], outfilename=self.output)
+        self.check_file_contains(self.output, self.key_to_json(self.string_to_unicode("key")))
+        self.check_file_contains(self.output, self.value_to_json(self.string_to_unicode("value")))
+        for i in range(1, 5):
+            self.check_file_not_contains(self.output, self.key_to_json(self.string_to_unicode(str(i))))
+            self.check_file_not_contains(self.output, self.key_to_json(self.string_to_unicode(str(i))))
+
+        # Call dump with -j and -k options missing key
+        self.runWt(['dump', '-j', '-k', "table", uri], outfilename=self.output)
+        self.check_file_not_contains(self.output, self.key_to_json(self.string_to_unicode("key")))
+        self.check_file_not_contains(self.output, self.value_to_json(self.string_to_unicode("value")))
+        for i in range(1, 5):
```

Comment:
It seems there are some test checking similarities between each dump call. Can we make it into a function? Also it would be worth testing out a scenario where -k matches a particular key.

Reply:
Changed in most recent commit